vehicle/src/Vehicle/Data/Builtin/Standard/Type.hs

```diff
@@ -188,10 +191,29 @@ restrictStandardDeclType declSort (ident, p) typ = do
         RestrictedNetwork -> ValidNetworkType
 
   let expr = BuiltinExpr p (TypeClass tc) [explicit typ]
-  let origin = InstanceTypeRestrictionOrigin $ TypeRestrictionOrigin env (ident, provenanceOf typ) declSort typ
+  let origin = InstanceTypeRestrictionOrigin $ TypeRestrictionOrigin env (ident, provenanceOf typ) (Left declSort) typ
   _ <- createFreshInstanceConstraint False mempty p origin Irrelevant expr
   return typ
 
+restrictStandardRecordType ::
+  forall m.
+  (MonadTypeChecker Builtin m) =>
+  DeclProvenance ->
+  Type Builtin ->
+  [RecordField (Type Builtin)] ->
+  m (Type Builtin)
+restrictStandardRecordType (_ident, _p) typ [] = return typ
+restrictStandardRecordType (ident, p) typ ((firstFieldName, firstFieldType) : restFields) = do
+  env <- getFreeEnv
+  let expr = BuiltinExpr p (TypeClass ValidTensorLikeType) [explicit firstFieldType]
+  let origin = InstanceTypeRestrictionOrigin $ TypeRestrictionOrigin env ((Identifier (modulePath ident) (nameOf firstFieldName)), provenanceOf firstFieldName) (Right RestrictedRecord) firstFieldType
```

Comment:
This is a very long line. Could you split some of it out into intermediate variables?